src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleThrowsTest.java

```diff
@@ -13,7 +13,7 @@
  * @author kevin.w.wall@gmail.com
  * @since 2019
  */
-package org.owasp.esapi.reference;
+package org.owasp.esapi.reference.validation;
```

Comment:
Good catch!

Reply:
So how does this test along test your changes? I think you need to be able to test:

1. The old test case that was being used prior to your changes.
2. Test your changes, both with a success case and a failure case. Ideally the success case should test all the successful possibilities (or maybe a different path for each different class loader).

Reply:
The JUnit test cases for this has been moved to the new file "src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleClasspathTest.java". However, problems still remain.
The following still needs to be addressed:
1. The class Javadoc, lines 34-49, is incorrect. It was blindly copied from HTMLValidationRulesThrowsTest.java and thus it needs changed accordingly.
2. A new test case needs to be added to confirm that rolling back to the default file (antisamy-esapi.xml) when the property specifies a new, but non-existing AntiSamy policy file name (or resource) cannot be found.
3. There needs to be a NEW AntiSamy test added that will ONLY work with your new "src/test/resources/antisamy-esapi-CP.xml" file. This would be to ensure that your changes to HTMLValidationRule.java don't get accidentally rolled back to what it was prior to your commit and then the default "src/test/resources/antisamy-esapi.xml" file just gets picked up and used. If those files are identical and every test works in both of these AntiSamy policy files, then we would not be able to determine that as all the tests in HTMLValidationClasspathTest.java would still pass. For that reason, the antisamy-esapi-CP.xml and antisamy-esapi.xml files should be different. (Please leave antisamy-esapi.xml as is and change antisamy-esapi-CP.xml.)

Reply:
Okay; ultimately it would be nice to have more complete testing here (i.e., testing all 3 of the class loaders rather than just the 'current thread context class loader'), but I understand how hard this is to mock test outside of a application server, so I am find with what you have here. I've at least confirmed that the case using the 'current thread context class loader' seems to be working properly. No further changes required here.